exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricConfiguration.java

```
@@ -19,86 +19,88 @@
-        return new AutoValue_MetricConfiguration.Builder()
-            .setProjectId(DEFAULT_PROJECT_ID)
-            .setDeadline(DEFAULT_DEADLINE);
-    }
-
-    /**
-     * Builder for {@link MetricConfiguration}.
-     */
-    @AutoValue.Builder
-    public abstract static class Builder {
+    public abstract Builder setProjectId(String projectId);
 
-        Builder() {
-        }
+    public abstract Builder setCredentials(Credentials newCredentials);
 
-        abstract String getProjectId();
 
-        abstract Duration getDeadline();
+    public abstract Builder setMetricServiceStub(MetricServiceStub newMetricServiceStub);
 
-        public abstract Builder setProjectId(String projectId);
+    public abstract Builder setDeadline(Duration deadline);
 
-        public abstract Builder setCredentials(Credentials newCredentials);
+    abstract MetricConfiguration autoBuild();
 
-        public abstract Builder setMetricServiceStub(MetricServiceStub newMetricServiceStub);
-
-        public abstract Builder setDeadline(Duration deadline);
-
-        abstract MetricConfiguration autoBuild();
-
-        /**
-         * Builds a {@link MetricConfiguration}.
-         *
-         * @return a {@code MetricsConfiguration}.
-         */
-        public MetricConfiguration build() {
-            Preconditions.checkArgument(
-                !Strings.isNullOrEmpty(getProjectId()),
-                "Cannot find a project ID from either configurations or application default.");
-            Preconditions.checkArgument(getDeadline().compareTo(ZERO) > 0, "Deadline must be positive.");
-            return autoBuild();
-        }
+    /**
+     * Builds a {@link MetricConfiguration}.
+     *
+     * @return a {@code MetricsConfiguration}.
+     */
+    public MetricConfiguration build() {
+      Preconditions.checkArgument(
+          !Strings.isNullOrEmpty(getProjectId()),
+          "Cannot find a project ID from either configurations or application default.");
```

Shall we return this error message right after the lookup?
Here we should just return a message that project ID is mandatory, I think.

---

As discussed, this is because the "default project ID" is part of the builder: https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/0896e0dce64f836f7587d8d02877dd4e9225fca4/exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricConfiguration.java#L63

So no change is needed, apologies.